code/datums/components/label.dm

```diff
@@ -0,0 +1,49 @@
+/**
+	The label component.
+
+	This component is used to manage labels applied by the hand labeler.
+
+	Atoms can only have one instance of this component, and therefore only one label at a time.
+	This is to avoid having names like "Backpack (label1) (label2) (label3)". This is annoying and abnoxious to read.
+
+	When a player clicks the atom with a hand labeler to apply a label, this component gets applied to it.
+	If the labeler is off, the component will be removed from it, and the label will be removed from its name.
+ */
+/datum/component/label
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	/// The name of the label the player is applying to the parent.
+	var/label_name
+
+/datum/component/label/Initialize(_label_name)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	label_name = _label_name
+
+	// Add the label to the name of the object in the format of: "Item name (label)"
+	var/atom/owner = parent
+	owner.name += " ([label_name])"
+
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/RemoveLabel)
```

Comment:
You should do these in
http://codedocs.tgstation13.org/datum/component.html#proc/RegisterWithParent

So the component can be transferred

Reply:
Alright, I'll do that, same with the unregisters